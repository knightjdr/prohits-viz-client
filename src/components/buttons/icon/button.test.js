import React from 'react';
import { render } from '@testing-library/react';
import { faPlus } from '@fortawesome/pro-solid-svg-icons';

import Button from './button';

const onClick = jest.fn();

const theme = {
  colorPrimary1: '#00f',
  fontLight: '#fff',
  timingFunction: 'ease',
};

const renderElement = props => render(<Button {...props} />);

describe('Icon button', () => {
  describe('style', () => {
    describe('primary button', () => {
      const props = {
        'aria-label': 'test-button',
        fill: 'primary',
        onClick,
        icon: faPlus,
        theme,
        type: 'button',
      };

      it('should match snapshot', () => {
        const { container } = renderElement(props);
        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('transparent (default) button', () => {
      const props = {
        'aria-label': 'test-button',
        onClick,
        icon: faPlus,
        theme,
        type: 'button',
      };

      it('should match snapshot', () => {
        const { container } = renderElement(props);
        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
