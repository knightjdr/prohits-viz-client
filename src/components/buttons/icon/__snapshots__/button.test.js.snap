// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon button style primary button should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: background-color,border-color,color;
  transition-property: background-color,border-color,color;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 34px;
  padding: 2px;
  position: relative;
  width: 34px;
  z-index: 1;
}

.c1::before {
  border-radius: 50%;
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transition-property: background-color,border-color,color,opacity,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,opacity,transform;
  transition-property: background-color,border-color,color,opacity,transform;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 100%;
  z-index: -1;
}

.c1 {
  background-color: transparent;
  border: none;
}

.c1::before {
  background-color: #00f;
  opacity: 0;
}

.c1:focus:not([disabled]),
.c1:hover:not([disabled]) {
  box-shadow: none;
  color: #fff;
}

.c1:focus:not([disabled])::before,
.c1:hover:not([disabled])::before {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  aria-label="test-button"
  class="c0 c1"
  fill="primary"
  kind="transparent"
  type="button"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-plus fa-w-14 fa-1x "
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`Icon button style transparent (default) button should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: background-color,border-color,color;
  transition-property: background-color,border-color,color;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-width: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 34px;
  padding: 2px;
  position: relative;
  width: 34px;
  z-index: 1;
}

.c1::before {
  border-radius: 50%;
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transition-property: background-color,border-color,color,opacity,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,opacity,transform;
  transition-property: background-color,border-color,color,opacity,transform;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 100%;
  z-index: -1;
}

.c1 {
  background-color: transparent;
  border: none;
}

.c1::before {
  background-color: #00f;
  opacity: 0;
}

.c1:focus:not([disabled]),
.c1:hover:not([disabled]) {
  box-shadow: none;
  color: #fff;
}

.c1:focus:not([disabled])::before,
.c1:hover:not([disabled])::before {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  aria-label="test-button"
  class="c0 c1"
  kind="transparent"
  type="button"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-plus fa-w-14 fa-1x "
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
    />
  </svg>
</button>
`;
